applications/poisson/sine/application.h

```diff
@@ -185,24 +185,24 @@ class Application : public ApplicationBase<dim, n_components, Number>
     unsigned int n_cells_1d =
       std::max((unsigned int)2, this->param.grid.n_subdivisions_1d_hypercube);
 
-    if(this->param.grid.element_type == ElementType::Hypercube)
-    {
-      dealii::GridGenerator::subdivided_hyper_cube(*this->grid->triangulation,
-                                                   n_cells_1d,
-                                                   left,
-                                                   right);
-    }
-    else if(this->param.grid.element_type == ElementType::Simplex)
-    {
-      dealii::GridGenerator::subdivided_hyper_cube_with_simplices(*this->grid->triangulation,
-                                                                  n_cells_1d,
-                                                                  left,
-                                                                  right);
-    }
-    else
-    {
-      AssertThrow(false, ExcNotImplemented());
-    }
+    auto const lambda_create_coarse_triangulation = [&](dealii::Triangulation<dim, dim> & tria) {
+      if(this->param.grid.element_type == ElementType::Hypercube)
+      {
+        dealii::GridGenerator::subdivided_hyper_cube(tria, n_cells_1d, left, right);
+      }
+      else if(this->param.grid.element_type == ElementType::Simplex)
+      {
+        dealii::GridGenerator::subdivided_hyper_cube_with_simplices(tria, n_cells_1d, left, right);
+      }
+      else
+      {
+        AssertThrow(false, ExcNotImplemented());
+      }
+    };
+
+    this->grid->create_triangulation(this->param.grid,
+                                     lambda_create_coarse_triangulation,
+                                     this->param.grid.n_refine_global);
```

Comment:
According to my current understanding, this change here is problematic in terms of the bug discussed in issue #313 because the part with Neumann boundary conditions below is done after this function call, which is not allowed, because the `coarse_grid_triangulations` for global-coarsening multigrid then have the wrong boundary conditions. So I conclude the bug has not been introduced by changes in #269, but PR #269 did not update this particular test case (poisson/sine) so that it can be used for global coarsening multigrid.

Reply:
I am looking into this!

Reply:
I moved the Neumann part into the lambda and it works. Im already working on it.

Reply:
Okay. Thanks!

Reply:
The problem is that - by the current design - we can not solve the problem with curvilinear mesh in poisson/sine correctly, see issue #319.